ui/redux/actions/memberships.js

```diff
@@ -0,0 +1,358 @@
+      const subscriptionFailedBackendError = 'failed to create subscription with default card';
+
+      // wait a bit to show the message so it's not jarring for the user
+      let errorMessageTimeout = 1150;
+
+      // don't do an error delay if there's already a network error
+      if (errorMessage === subscriptionFailedBackendError) {
+        errorMessageTimeout = 0;
+      }
+
+      setTimeout(() => {
+        const genericErrorMessage = __(
+          "Sorry, your purchase wasn't able to completed. Please contact support for possible next steps"
+        );
+
+        dispatch(doToast({ message: genericErrorMessage, isError: true }));
+      }, errorMessageTimeout);
+
+      throw new Error(e);
+    });
+};
+
+export const doMembershipCancelForMembershipId = (membershipId: number) => async (dispatch: Dispatch) => {
+  dispatch({ type: ACTIONS.SET_MEMBERSHIP_CANCEL_STARTED, data: membershipId });
+
+  return await Lbryio.call(
+    'membership',
+    'cancel',
+    { environment: stripeEnvironment, membership_id: membershipId },
+    'post'
+  )
+    .then((response) => {
+      dispatch({ type: ACTIONS.SET_MEMBERSHIP_CANCEL_SUCCESFUL, data: membershipId });
+      dispatch(doMembershipMine());
+
+      return response;
+    })
+    .catch((e) => {
+      dispatch({ type: ACTIONS.SET_MEMBERSHIP_CANCEL_FAILED, data: membershipId });
+      throw new Error(e);
+    });
+};
+
+export const doMembershipAddTier = (params: MembershipAddTierParams) => async (dispatch: Dispatch) =>
+  await Lbryio.call('membership', 'add', { ...params, environment: stripeEnvironment }, 'post');
+
+export const doGetMembershipPerks = (params: MembershipListParams) => async (dispatch: Dispatch) =>
+  await Lbryio.call(
+    'membership_perk',
+    'list',
+    { ...params, environment: stripeEnvironment },
+    'post'
+  ).then((response: MembershipDetails) => dispatch({ type: ACTIONS.MEMBERSHIP_PERK_LIST_COMPLETE, data: response }));
+
+export const doOpenCancelationModalForMembership = (membership: MembershipTier) => (
+  dispatch: Dispatch,
+  getState: GetState
+) => {
+  const { MembershipDetails, Subscription } = membership;
+
+  const state = getState();
+  const formattedEndOfMembershipDate = formatDateToMonthDayAndYear(Subscription.current_period_end * 1000);
+  const creatorUri = buildURI({
+    channelName: MembershipDetails.channel_name,
+    channelClaimId: MembershipDetails.channel_id,
+  });
+  const creatorTitleName = selectChannelTitleForUri(state, creatorUri);
+
+  return dispatch(
+    doOpenModal(MODALS.CONFIRM, {
+      title: __('Confirm %membership_name% Cancelation', { membership_name: MembershipDetails.name }),
+      subtitle: __(
+        'Are you sure you want to cancel your %creator_name%\'s "%membership_name%" membership? ' +
+          'You will still have all your features until %end_date% at which point your purchase will not renewed ' +
+          'and you will lose access to your membership features and perks',
+        {
```

Comment:
:small_red_triangle: Missing `be` in `will not renewed`?
:small_red_triangle:  End sentence with punctuation.